.github/workflows/node.js.yaml

```diff
@@ -0,0 +1,66 @@
+name: Node.js CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+      # - name: Install node modules
+        # run: npm ci
+
+      # - name: Get npm cache directory
+        # id: npm-cache-dir
+        # shell: bash
+        # run: echo "dir=$(npm config get cache)" >> ${GITHUB_OUTPUT}
+
+      - name: Restore node_modules cache
+        id: node_modules-cache-restore
+        uses: actions/cache/restore@v3
```

Comment:
I've never used this.  If you think it will work, let's try.

However, we can also do this later if you want to keep things simple and always install the deps in every job